stablehlo/tests/infer_chlo.mlir

```diff
@@ -120,10 +120,14 @@ func.func @broadcast_select_branch_mismatch(%arg0: tensor<2xi1>, %arg1: tensor<2
 // -----
 // CHECK-LABEL: @broadcast_select_reify
 func.func @broadcast_select_reify(%arg0: tensor<2xi1>, %arg1: tensor<?xi32>, %arg2: tensor<?xi32>) -> tensor<1xindex> {
-  // CHECK:      %0 = shape.const_shape [2] : tensor<1xindex>
-  // CHECK-NEXT: %1 = shape.shape_of %arg1 : tensor<?xi32> -> tensor<1xindex>
-  // CHECK-NEXT: %2 = shape.shape_of %arg2 : tensor<?xi32> -> tensor<1xindex>
-  // CHECK-NEXT: %3 = shape.broadcast %1, %2, %0 : tensor<1xindex>, tensor<1xindex>, tensor<1xindex> -> tensor<1xindex>
+  // CHECK-DAG: %[[ARG0_S:.+]] = shape
+  // CHECK-DAG: %[[ARG1_S:.+]] = shape
+  // CHECK-DAG: %[[ARG2_S:.+]] = shape
```

Comment:
Ah. I see what you're saying now. Hmm, this is because we use `return builder.createOrFold<shape::ShapeOfOp>(loc, v);` instead of just `create`. Meaning as it's written, the fold semantics of shape op should be a part of this test unfortunately.. I.e. if fold changed in a way that broke this lowering, we'd want to know.

Reply:
Gotcha. This will happen in the next LLVM integrate: they removed the folding so we will get `shape_of` now. Should I just drop this change then? (Or maybe we can keep it open until the integrate in case we need to refer to it.)